### ISSUE TITLE
Gesture typing is not working anymore

### ISSUE THREAD

**Issue by vit1-irk** [termux/termux-app#230]
I have updated Termux a while ago and now it's seen that you can not use the fast gesture typing (on-screen keyboard) anymore.

This feature was really useful to enter some text in Vim and even in regular command line. It takes more time to perform input by one symbol.

Please add the way to re-enable it somehow or take the old behavior back.

Thanks.

**Comment by Neo-Oli** [comment]
Are you sure you're using the Text input view? https://termux.com/touch-keyboard.html#extra-keys-view

**Comment by vit1-irk** [comment]
Yes, I know about this feature, but you need to switch between those Alt, Tab, Ctrl, etc touch keys and the Text input view all the time if you are working with Vim or other interactive application.

Maybe it is possible to combine extra keys and input view without scrolling?

**Comment by Neo-Oli** [comment]
Gesture typing requires a text input field that is able to communicate back to the keyboard application, so that the keyboard knows what has already been typed. As far as I am aware this would be impossible to implement into a terminal emulator.

**Comment by vit1-irk** [comment]
In previous versions of Termux combined typing worked, i remember clearly.

But now you can alternatively add the feature of keeping the extra keys bar and Text input view in the same screen.

Example algorithm:

1. User is typing text in Text input view
2. Press special key (Ctrl, Alt, Escape)
3. Entered text transfers into application
4. The special key is kept to complete the key combination
5. Keybinding is processed by console app
6. Termux input focus goes back to Text input view

**Comment by eboody** [comment]
https://wiki.termux.com/wiki/Touch_Keyboard#Text_Input_View

Text Input View

Terminal emulators usually do not support the advanced features of touch keyboards like autocorrect, prediction and swipe typing. To solve this, Termux has a text input view. Text entered in it will get pasted to the terminal. Because it's a native Android text input view, all touch keyboard features will work. To access the text input view you have to swipe the extra keys view to the left.

**Comment by tribixbite** [comment]
https://github.com/user-attachments/assets/10955c66-81b0-4dd2-b3de-05d718fc6585

**Comment by JoakimCh** [comment]
> Screen_Recording_20251028_232539_Termux.mp4

What did you even do? I can't find a way to do that, nor any text input mode.

**Comment by agnostic-apollo** [comment]
https://wiki.termux.dev/wiki/Touch_Keyboard#Text_Input_View

**Comment by tribixbite** [comment]
> > Screen_Recording_20251028_232539_Termux.mp4
> 
> What did you even do? I can't find a way to do that, nor any text input mode.

I literally built my own gesture typing keyboard app specifically to solve this termux limitation lol.

it took a few months. I've continued to build it into the android kb I've wanted for ~18ish years...

https://github.com/tribixbite/CleverKeys

**Comment by agnostic-apollo** [comment]
lolz, this is the way :p
